### ISSUE TITLE
[bitnami/minio] TLS distributed installation fails

### ISSUE THREAD

**Issue by ml14tlc** [bitnami/charts#10307]
### Name and Version

bitnami/minio 11.3.5

### What steps will reproduce the bug?

Deploy a MinIO distributed cluster with TLS using a self-signed certificate

### Are you using any custom parameters or values?

Here's the `values` file

```yaml
global:
  imageRegistry: ""
  imagePullSecrets: []
  storageClass: ""
nameOverride: ""
fullnameOverride: ""
commonLabels: {}
commonAnnotations: {}
kubeVersion: ""
clusterDomain: cluster.local
extraDeploy: []
image:
  registry: docker.io
  repository: bitnami/minio
  tag: 2022.5.8-debian-10-r0
  pullPolicy: IfNotPresent
  pullSecrets: []
  debug: true
clientImage:
  registry: docker.io
  repository: bitnami/minio-client
  tag: 2022.5.4-debian-10-r4
mode: distributed
auth:
  existingSecret: "minio-root"
  forcePassword: false
  useCredentialsFiles: false
  forceNewKeys: false
defaultBuckets: ""
disableWebUI: false
tls:
  enabled: true
  existingSecret: dev-minio
  mountPath: /.mc/certs
extraEnvVars:
  - name: MINIO_STORAGE_CLASS_STANDARD
    value: "EC:2"
  - name: MINIO_STORAGE_CLASS_RRS
    value: "EC:2"
extraEnvVars: {}
extraEnvVarsCM: ""
extraEnvVarsSecret: ""
command: []
args: []
schedulerName: ""
deployment:
  updateStrategy:
    type: Recreate
statefulset:
  updateStrategy:
    type: RollingUpdate
  podManagementPolicy: Parallel
  replicaCount: 4
  zones: 1
  drivesPerNode: 1
provisioning:
  enabled: true
  schedulerName: ""
  podAnnotations: {}
  command: []
  args: []
  extraVolumes: []
  extraVolumeMounts: []
  resources:
    limits: {}
    requests: {}
  policies: []
  users: []
  groups: []
  buckets: []
hostAliases: []
containerPorts:
  api: 9000
  console: 9001
podSecurityContext:
  enabled: true
  fsGroup: 1001
containerSecurityContext:
  enabled: true
  runAsUser: 1001
  runAsNonRoot: true
podLabels: {}
podAnnotations: {}
podAffinityPreset: ""
podAntiAffinityPreset: soft
nodeAffinityPreset:
  type: ""
  key: ""
  values: []
affinity: {}
nodeSelector: {}
tolerations: []
topologySpreadConstraints: []
priorityClassName: ""
resources:
  limits: {}
  requests: {}
livenessProbe:
  enabled: true
  initialDelaySeconds: 5
  periodSeconds: 5
  timeoutSeconds: 5
  successThreshold: 1
  failureThreshold: 5
readinessProbe:
  enabled: true
  initialDelaySeconds: 5
  periodSeconds: 5
  timeoutSeconds: 1
  successThreshold: 1
  failureThreshold: 5
startupProbe:
  enabled: false
  initialDelaySeconds: 0
  periodSeconds: 10
  timeoutSeconds: 5
  successThreshold: 1
  failureThreshold: 60
customLivenessProbe: {}
customReadinessProbe: {}
customStartupProbe: {}
lifecycleHooks: {}
extraVolumes: []
extraVolumeMounts: []
initContainers: []
sidecars: []
service:
  type: ClusterIP
  ports:
    api: 9000
    console: 9001
  nodePorts:
    api: ""
    console: ""
  clusterIP: ""
  loadBalancerIP: ""
  loadBalancerSourceRanges: []
  externalTrafficPolicy: Cluster
  extraPorts: []
  annotations: {}
ingress:
  enabled: false
  apiVersion: ""
  ingressClassName: ""
  hostname: minio.local
  path: /
  pathType: ImplementationSpecific
  servicePort: minio-console
  annotations: {}
  tls: false
  selfSigned: false
  extraHosts: []
  extraPaths: []
  extraTls: []
  secrets: []
apiIngress:
  enabled: false
  apiVersion: ""
  ingressClassName: ""
  hostname: minio.local
  path: /
  pathType: ImplementationSpecific
  servicePort: minio-api
  annotations: {}
  tls: false
  selfSigned: false
  extraHosts: []
  extraPaths: []
  extraTls: []
  secrets: []
networkPolicy:
  enabled: false
  allowExternal: true
  extraFromClauses: {}
persistence:
  enabled: true
  storageClass: hcloud-volumes
  mountPath: /data
  accessModes:
    - ReadWriteOnce
  size: 1Gi
  annotations: {}
  existingClaim: ""
volumePermissions:
  enabled: false
  image:
    registry: docker.io
    repository: bitnami/bitnami-shell
    tag: 10-debian-10-r420
    pullPolicy: IfNotPresent
    pullSecrets: []
  resources:
    limits: {}
    requests: {}
  containerSecurityContext:
    runAsUser: 0
serviceAccount:
  create: true
  name: ""
  automountServiceAccountToken: true
  annotations: {}
pdb:
  create: false
  minAvailable: 1
  maxUnavailable: ""
metrics:
  prometheusAuthType: public
  serviceMonitor:
    enabled: false
    namespace: monitoring
    labels: {}
    jobLabel: ""
    path: /minio/v2/metrics/cluster
    interval: 30s
    scrapeTimeout: ""
    metricRelabelings: []
    relabelings: []
    honorLabels: false
    selector: {}
gateway:
  enabled: false
  type: s3
  replicaCount: 4
  updateStrategy:
    type: Recreate
  autoscaling:
    enabled: false
    minReplicas: "4"
    maxReplicas: "4"
    targetCPU: ""
    targetMemory: ""
  priorityClassName: ""
  auth:
    azure:
      accessKey: ""
      secretKey: ""
      serviceEndpoint: ""
      storageAccountName: ""
      storageAccountKey: ""
      storageAccountNameExistingSecret: ""
      storageAccountNameExistingSecretKey: ""
      storageAccountKeyExistingSecret: ""
      storageAccountKeyExistingSecretKey: ""
    gcs:
      accessKey: ""
      secretKey: ""
      keyJSON: ""
      projectID: ""
    nas:
      accessKey: ""
      secretKey: ""
    s3:
      accessKey: ""
      secretKey: ""
      serviceEndpoint: https://s3.amazonaws.com
```

### What is the expected behavior?

MinIO cluster starts up and it uses `https`

### What do you see instead?

Any MinIO node shows the following message

```
ERROR Unable to start the server: HTTPS specified in endpoints, but no TLS certificate is found on the local machine
      > Please add TLS certificate or use HTTP endpoints only
      HINT:
        Refer to https://docs.min.io/docs/how-to-secure-access-to-minio-server-with-tls for information about how to load a TLS certificate in your server
```

### Additional information

Searching online I found [this](https://github.com/minio/operator/issues/211) issue, but I'm using the latest version of the chart (at the time of writing) and I have Kubernetes `1.22.6` on Ubuntu `20.04`.

The certificate I'd like to use for MinIO has been generated by the `cert manager` and it's self-signed. The secret (`dev-minio` in the config) exists and it contains the following cert

```
subject=CN = dev-minio
issuer=CN = dev-cluster-ca
notBefore=May 18 15:26:33 2022 GMT
notAfter=May 18 15:26:33 2023 GMT
```

The manifest of the generated pod

```yaml
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2022-05-18T15:11:46Z"
  generateName: dev-minio-
  labels:
    app.kubernetes.io/instance: dev
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: minio
    controller-revision-hash: dev-minio-57c4d6b4d4
    helm.sh/chart: minio-11.3.5
    statefulset.kubernetes.io/pod-name: dev-minio-0
  name: dev-minio-0
  namespace: minio
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: StatefulSet
    name: dev-minio
    uid: 6814eab6-a26f-47a2-b65a-1cbf3fa048b7
  resourceVersion: "12465743"
  uid: 4a16f2c4-3e7f-46b8-ae6a-fd4878958abd
spec:
  affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - podAffinityTerm:
          labelSelector:
            matchLabels:
              app.kubernetes.io/instance: dev
              app.kubernetes.io/name: minio
          namespaces:
          - minio
          topologyKey: kubernetes.io/hostname
        weight: 1
  containers:
  - env:
    - name: BITNAMI_DEBUG
      value: "true"
    - name: MINIO_DISTRIBUTED_MODE_ENABLED
      value: "yes"
    - name: MINIO_DISTRIBUTED_NODES
      value: dev-minio-{0...3}.dev-minio-headless.minio.svc.cluster.local:9000/data
    - name: MINIO_SCHEME
      value: https
    - name: MINIO_FORCE_NEW_KEYS
      value: "no"
    - name: MINIO_ROOT_USER
      valueFrom:
        secretKeyRef:
          key: root-user
          name: minio-root
    - name: MINIO_ROOT_PASSWORD
      valueFrom:
        secretKeyRef:
          key: root-password
          name: minio-root
    - name: MINIO_SKIP_CLIENT
      value: "yes"
    - name: MINIO_BROWSER
      value: "on"
    - name: MINIO_PROMETHEUS_AUTH_TYPE
      value: public
    - name: MINIO_CERTSDIR
      value: /.mc/certs
    - name: MINIO_CONSOLE_PORT_NUMBER
      value: "9001"
    image: docker.io/bitnami/minio:2022.5.8-debian-10-r0
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 5
      httpGet:
        path: /minio/health/live
        port: minio-api
        scheme: HTTPS
      initialDelaySeconds: 5
      periodSeconds: 5
      successThreshold: 1
      timeoutSeconds: 5
    name: minio
    ports:
    - containerPort: 9000
      name: minio-api
      protocol: TCP
    - containerPort: 9001
      name: minio-console
      protocol: TCP
    readinessProbe:
      failureThreshold: 5
      initialDelaySeconds: 5
      periodSeconds: 5
      successThreshold: 1
      tcpSocket:
        port: minio-api
      timeoutSeconds: 1
    resources: {}
    securityContext:
      runAsNonRoot: true
      runAsUser: 1001
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /.mc/certs
      name: minio-certs
    - mountPath: /data
      name: data
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-g2w79
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  hostname: dev-minio-0
  nodeName: dev-k8s-wrk01
  preemptionPolicy: PreemptLowerPriority
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext:
    fsGroup: 1001
  serviceAccount: dev-minio
  serviceAccountName: dev-minio
  subdomain: dev-minio-headless
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - name: data
    persistentVolumeClaim:
      claimName: data-dev-minio-0
  - name: minio-certs
    secret:
      defaultMode: 420
      items:
      - key: tls.crt
        path: public.crt
      - key: tls.key
        path: private.key
      - key: ca.crt
        path: CAs/public.crt
      secretName: dev-minio
  - name: kube-api-access-g2w79
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
```

**Comment by rafariossaa** [comment]
Hi,
  I am creating an internal task to look further into this.
  We will come back as soon as we have news.

**Comment by ml14tlc** [comment]
hi @rafariossaa 

thanks a lot for the support. I was able to start the cluster in secure mode passing the following `command`: ``["/opt/bitnami/minio/bin/minio", "server", "--certs-dir", "/.mc/certs", "--console-address", ":9001", "--address", ":9000", "https://dev-minio-{0...3}.dev-minio-headless.minio.svc.cluster.local:9000/data"]`, where `/.mc/certs` is the folder mounted directly from the cert's secret. If I'm not wrong, the image's entrypoint and the subsequents scripts are copying the certs somewhere else. Maybe there's something going wrong in the process.
`

**Comment by ml14tlc** [comment]
hi again @rafariossaa 

I think I have a better understanding now:

* I did _not_ specify `tls.mountPath`, it prevents the MinIO servers to start, since `MINIO_CERTS_DIR` still points to `/certs`
* unfortunately, the provisioning job fails, it complains that the servers' cert comes from an unknown authority (it's self signed). This is the reason why I set `tls.mountPath` to `/.mc/certs` (since I'm using an unprivileged user, this amounts to `$HOME/.mc/certs`, i.e., the default location)

In order to solve the problem with the provisioner, I mounted the secret with the self signed cert as extra volume, i.e., 

```
provisioning:
  extraVolumes:
    - name: minio-client-certs
      secret:
        secretName: dev-minio
        items:
          - key: ca.crt
            path: CAs/public.crt
  extraVolumeMounts:
    - name: minio-client-certs
      mountPath: /.mc/certs
      readOnly: true
```

where `dev-minio` is the same value assigned to `tls.existingSecret`.

Hope this helps.

**Comment by carrodher** [comment]
Thanks for the follow-up, did you solve the issue with the latest changes?

I'm sharing this info in the internal task, as well as the recently created ticket we closed as duplicated.

**Comment by ml14tlc** [comment]
Hi @carrodher ,

yes, I was able to start a distributed cluster using a self signed certificate for TLS simply avoiding `tls.mountPath` and adding the extra volume to the provisioner pod as described in the previous post.

**Comment by rafariossaa** [comment]
Hi,
  Thanks for the info.

**Comment by FraPazGal** [comment]
Hello @ml14tlc,

We have just merged #10541 and updated the container's logic in https://github.com/bitnami/bitnami-docker-minio/commit/2bc8c0ef0aad52c5d1904b6bc655b1a228f28716. TLS configuration should be working as expected now but do not doubt to reply here if you encounter any other issues.
